app/controllers/api/v4/identifiers_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Api::V4::IdentifiersController < UnauthenticatedApiController
+  include CorsSupport
+
+  def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+    build_path = AppVersionBuildPath.new(build_params)
+    if build_path.valid?
+      claim = IdentifierClaim.new(create_params)
+      if claim.save!
+        visitor = claim.identifier.visitor
+        app_build = build_path.app_build
+        @active_splits = Split.for_presentation(app_build: app_build)
+        @visitor_id = visitor.id
+        @assignments = visitor.assignments_for(app_build).includes(:split).order(:updated_at)
+        @experience_sampling_weight = Rails.configuration.experience_sampling_weight
+      else
+        render_errors claim
+      end
+    else
+      render_errors build_path
+    end
+  end
```

Comment:
this is basically the max logic i'm okay with putting in a controller. creating a new model for this just feels unnecessary to me.

Reply:
I think there could be a case for it if we changed the JSON structural representation in a future version of this endpoint, then we'd have exactly the same controller code copypasta'd. Does that move you?

Reply:
it doesn't really move me 😄 

my vote is to refactor this into a model as a part of that future work to avoid the copypasta because at least then we'd know that we're actually reusing the same logic in more than one place. but really i'm just trying to avoid having to create a new model that figures out how to coordinate between these two different models right now 😛

Reply:
ah, F-it. i decided to just dive into this.